comp/core/workloadmeta/collectors/internal/docker/docker.go

```diff
@@ -50,6 +50,83 @@ const imageEventActionSbom = events.Action("sbom")
 
 type resolveHook func(ctx context.Context, co types.ContainerJSON) (string, error)
 
+type flavors struct {
+	mutex               sync.Mutex
+	containerIds        map[string]struct{}
+	containerIdToFlavor map[string]string
+}
+
+func NewFlavors() flavors {
+	return flavors{
+		containerIds:        make(map[string]struct{}),
+		containerIdToFlavor: make(map[string]string),
+	}
+}
+
+func (f *flavors) AddContainer(containerId string) {
+	log.Infof("flavors: add container %s", containerId)
+	f.mutex.Lock()
+	defer f.mutex.Unlock()
+	f.containerIds[containerId] = struct{}{}
+}
+
+func (f *flavors) RemoveContainer(containerId string) {
+	log.Infof("flavors: remove container %s", containerId)
+	f.mutex.Lock()
+	defer f.mutex.Unlock()
+	delete(f.containerIds, containerId)
+	delete(f.containerIdToFlavor, containerId)
+}
+
+func (f *flavors) GetFlavor(containerId string) string {
+	f.mutex.Lock()
+	defer f.mutex.Unlock()
+	return f.containerIdToFlavor[containerId]
+}
+
+func (f *flavors) Run(ctx context.Context, c *collector) {
+	for {
+		if err := ctx.Err(); err != nil {
+			return
+		}
+		time.Sleep(time.Second)
+
+		f.mutex.Lock()
+		containerIds := make([]string, 0, len(f.containerIds))
```

Comment:
debatable/not important, but since I think the common case is you don't end up filling this, there's no need to preallocate capacity
```suggestion
		containerIds := make([]string, 0)
```

Reply:
Done.